File: Desktop/app flutter/projet-etudiants/api-spring-boot/src/main/java/com/etudiants/api/entity/Etudiant.java
package com.etudiants.api.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonFormat;
import java.time.LocalDate;

@Entity
@Table(name = "etudiants")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Etudiant {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "cin", nullable = false, unique = true)
    private String cin;
    
    @Column(name = "nom", nullable = false)
    private String nom;
    
    @Column(name = "date_naissance")
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate dateNaissance;

    @Column(name = "departement")
    private String departement;
}

File: Desktop/app flutter/projet-etudiants/mobile-app/lib/models/etudiant.dart
class Etudiant {
  final int id;
  final String cin;
  final String nom;
  final String? dateNaissance;
  final String? departement;

  Etudiant({
    required this.id,
    required this.cin,
    required this.nom,
    this.dateNaissance,
    this.departement,
  });

  factory Etudiant.fromJson(Map<String, dynamic> json) {
    return Etudiant(
      id: json['id'],
      cin: json['cin'],
      nom: json['nom'],
      dateNaissance: json['dateNaissance'],
      departement: json['departement'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'cin': cin,
      'nom': nom,
      'dateNaissance': dateNaissance,
      'departement': departement,
    };
  }
}
